esmvalcore/preprocessor/_time.py

```diff
@@ -553,3 +554,109 @@ def regrid_time(cube, frequency):
 
     return cube
+
+
+def low_pass_weights(window, cutoff):
+    """
+    Calculate weights for a low pass Lanczos filter.
+
+    Method borrowed from `iris example
+    <https://scitools.org.uk/iris/docs/latest/examples/General/
+    SOI_filtering.html?highlight=running%20mean>`_
+
+    Parameters
+    ----------
+    window: int
+        The length of the filter window.
+    cutoff: float
+        The cutoff frequency in inverse time steps.
+
+    Returns
+    -------
+    list:
+        List of floats representing the weights.
+    """
+    order = ((window - 1) // 2) + 1
+    nwts = 2 * order + 1
+    w = np.zeros([nwts])
+    n = nwts // 2
+    w[n] = 2 * cutoff
+    k = np.arange(1., n)
+    sigma = np.sin(np.pi * k / n) * n / (np.pi * k)
+    firstfactor = np.sin(2. * np.pi * cutoff * k) / (np.pi * k)
+    w[(n - 1):0:-1] = firstfactor * sigma
+    w[(n + 1):-1] = firstfactor * sigma
+
+    return w[1:-1]
+
+
+def timeseries_filter(cube, window, span,
+                      filter_type='lowpass', filter_stats='sum'):
+    """
+    Apply a timeseries filter.
+
+    Method borrowed from `iris example
+    <https://scitools.org.uk/iris/docs/latest/examples/General/
+    SOI_filtering.html?highlight=running%20mean>`_
+
+    Apply each filter using the rolling_window method used with the weights
+    keyword argument. A weighted sum is required because the magnitude of
+    the weights are just as important as their relative sizes.
+
+    See also the `iris rolling window
+    <https://scitools.org.uk/iris/docs/v2.0/iris/iris/
+    cube.html#iris.cube.Cube.rolling_window>`_
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        input cube.
+    window: int
+        The length of the filter window (in units of cube time coordinate).
+    span: int
+        Number of months/days (depending on data frequency) on which
+        weights should be computed e.g. 2-yearly: span = 24 (2 x 12 months).
+        Span should have same units as cube time coordinate.
+    filter_type: str, optional
+        Type of filter to be applied; default 'lowpass'.
+        Available types: 'lowpass'.
+    filter_stats: str, optional
+        Type of statistic to aggregate on the rolling window; default 'sum'.
+        Available operators: 'mean', 'median', 'std_dev', 'sum', 'min', 'max'
+
+    Returns
+    -------
+    iris.cube.Cube
+        cube time-filtered using 'rolling_window'.
+
+    Raises
+    ------
+    iris.exceptions.CoordinateNotFoundError:
+        Cube does not have time coordinate.
+    NotImplementedError:
+        If filter_type is not 'lowpass'.
+    """
+    try:
+        cube.coord('time')
+    except iris.exceptions.CoordinateNotFoundError:
+        logger.error("Cube %s does not have time coordinate", cube)
+        raise
+
+    # Construct weights depending on frequency
+    # TODO implement more filters!
+    supported_filters = ['lowpass', ]
+    if filter_type in supported_filters:
```

Comment:
You don't really need to have two levels of `if`.

Reply:
that assumes there are different functions for different filter types